engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsEventListener.java

```diff
@@ -0,0 +1,335 @@
+import java.time.ZoneOffset;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@ApplicationScoped
+public class ExportsEventListener {
+
+    public static final String APPLICATION_NAME = "urn:redhat:application:notifications";
+    public static final String CE_EXPORT_REQUEST_TYPE = "com.redhat.console.export-service.request";
+    public static final String EXPORT_CHANNEL = "export-requests";
+    public static final String EXPORT_SERVICE_URN = "urn:redhat:source:console:app:export-service";
+    public static final String FILTER_DATE_FROM = "from";
+    public static final String FILTER_DATE_TO = "to";
+    public static final String RESOURCE_TYPE_EVENTS = "urn:redhat:application:notifications:export:events";
+
+    protected static final String EXPORTS_SERVICE_FAILURES_COUNTER = "exports.service.failures";
+    protected static final String EXPORTS_SERVICE_SUCCESSES_COUNTER = "exports.service.successes";
+
+    private final ConsoleCloudEventParser consoleCloudEventParser = new ConsoleCloudEventParser();
+    private final Pattern subjectUuidExtractPattern = Pattern.compile("^urn:redhat:subject:export-service:request:(?<uuid>.+)$");
+
+    @Inject
+    EventRepository eventRepository;
+
+    @RestClient
+    ExportsService exportsService;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @Inject
+    StatelessSessionFactory statelessSessionFactory;
+
+    @ConfigProperty(name = "export-service.psk")
+    String exportServicePsk;
+
+    /**
+     * Listens to the exports channel, extracts the request's data, and if
+     * Notifications is the target application, and the requested resource and
+     * formats are supported, the corresponding payload is sent to the export
+     * service.
+     * @param message the incoming message from the channel.
+     * @return a read commit on the received message.
+     */
+    @Blocking
+    @Incoming(EXPORT_CHANNEL)
+    @Transactional
+    public CompletionStage<Void> eventListener(final Message<String> message) {
+        // If the integration is disabled simply ignore the messages.
+        if (!this.featureFlipper.isExportsServiceIntegrationEnabled()) {
+            return message.ack();
+        }
+
+        // Attempt deserializing the received message as a Cloud Event.
+        final ConsoleCloudEvent receivedEvent;
+        try {
+            receivedEvent = this.consoleCloudEventParser.fromJsonString(message.getPayload());
+        } catch (final ConsoleCloudEventParsingException e) {
+            Log.errorf("the received payload from the 'export-requests' topic is not a parseable Cloud Event: %s", e.getMessage());
+
+            this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
```

Comment:
WDYT of adding a tag to differentiate each failure case in the metrics?

Reply:
Oh crap. You said a tag. I've implemented and pushed different counters for the errors :man_facepalming: . Let me fix that...

Reply:
I am having an issue with this one. I have pushed the latest code to the PR. Essentially, I am unable to make the proper counting with the Micrometer Assertion Helper... I think I am passing the right tags with the correct values to assert, but I might be doing something wrong here?

It's the latest commit the one that has these changes.